private/couchbase/transactions/result.hxx

```diff
@@ -0,0 +1,267 @@
+/*
+ *     Copyright 2021-Present Couchbase, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+#pragma once
+
+#include "core/utils/json.hxx"
+
+#include <couchbase/transactions/internal/binary.hxx>
+
+#include <couchbase/codec/json_transcoder.hxx>
+
+#include <optional>
+#include <string>
+#include <vector>
+
+/**
+ * @file
+ * Provides an object encapsulating the results of a kv operation in Couchbase Transactions Client.
+ */
+namespace couchbase
+{
+namespace core::operations
+{
+struct lookup_in_response;
+struct mutate_in_response;
+} // namespace core::operations
+
+namespace transactions
+{
+struct result_base {
+    std::vector<std::byte> raw_value;
+    std::error_code ec;
+
+    result_base() = default;
+
+    result_base(const std::vector<std::byte>& res)
+      : raw_value(res)
+    {
+    }
+
+    bool has_value() const
+    {
+        return !raw_value.empty();
+    }
+};
+
+/**
+ * @brief Result of a subdoc operation.
+ *
+ * See @ref collection.lookup_in and @ref collection.mutate_in
+ */
+struct subdoc_result : result_base {
```

Comment:
needed because of the past (no cxx client so I had my own, etc...) Internally some blocking calls use it.   Can go away once we update those code paths.   Till then, it is private.